Wip/6.0 Various cosmetic code improvements

NathanQingyangXu opened the pull request:
The following obvious code improvements are done, including:

- unnecessary type specification (e.g. `Collections.<PackageDescriptor>emptySet()`);
- unnecessary nullness checking (e.g. `dataSourceConfigValue != null && String.class.isInstance( dataSourceConfigValue )`);
- unnecessary string creation in StringBuilder (e.g. `buffer.append( sql.substring( 0, pos ) )`);
- many others

Should be easy to review and verify.

NathanQingyangXu commented on hibernate-core/src/main/java/org/hibernate/boot/internal/InFlightMetadataCollectorImpl.java:
```diff
@@ -688,7 +688,7 @@ public void applyResultSetMapping(NamedResultSetMappingDefinition resultSetMappi
 		final String name = definition.getRegistrationName();
-		if ( !defaultSqlResultSetMappingNames.contains( name ) && sqlResultSetMappingMap.containsKey( name ) ) {
+		if ( !defaultSqlResultSetMappingNames.contains( name ) ) {
 			sqlResultSetMappingMap.remove( name );
```
we don't need key checking when we simply wanna remove it, regardless of whether it exists or not.

NathanQingyangXu commented on hibernate-core/src/main/java/org/hibernate/context/internal/ThreadLocalSessionContext.java:
```diff
@@ -111,7 +111,7 @@ private boolean needsWrapping(Session session) {
 		if ( Proxy.isProxyClass( session.getClass() ) ) {
 			final InvocationHandler invocationHandler = Proxy.getInvocationHandler( session );
-			if ( invocationHandler != null && TransactionProtectionWrapper.class.isInstance( invocationHandler ) ) {
+			if ( TransactionProtectionWrapper.class.isInstance( invocationHandler ) ) {
```
We don't need nullness checking for either `instanceof` or `Class#isInstance()` will take care of that already

NathanQingyangXu commented on hibernate-core/src/main/java/org/hibernate/dialect/Dialect.java:
```diff
@@ -3094,7 +3094,7 @@ public boolean supportsNoWait() {
 	 */
 	public String inlineLiteral(String literal) {
-		return String.format( "\'%s\'", escapeLiteral( literal ) );
+		return String.format( "'%s'", escapeLiteral( literal ) );
```
We don't need `'` escaping in Java String

NathanQingyangXu commented on hibernate-core/src/main/java/org/hibernate/dialect/function/TimestampaddFunction.java:
```diff
@@ -90,8 +90,7 @@ public SelfRenderingFunctionSqlAstExpression expression(
 		return new SelfRenderingFunctionSqlAstExpression(
-				(sqlAppender, sqlAstArguments1, walker)
-						-> render(sqlAppender, sqlAstArguments1, walker),
+				this::render,
```
JDK8's method reference can simplify our code clutter dramatically

NathanQingyangXu commented on hibernate-core/src/main/java/org/hibernate/internal/FastSessionServices.java:
```diff
@@ -221,11 +221,11 @@ private static boolean isTransactionAccessible(SessionFactoryImpl sf, Transactio
+		p.put( JPA_LOCK_SCOPE, PessimisticLockScope.EXTENDED.name() );
+		p.put( JPA_LOCK_TIMEOUT, LockOptions.WAIT_FOREVER );
+		p.put( JPA_SHARED_CACHE_RETRIEVE_MODE, CacheModeHelper.DEFAULT_RETRIEVE_MODE );
+		p.put( JPA_SHARED_CACHE_STORE_MODE, CacheModeHelper.DEFAULT_STORE_MODE );
```
Seems there is no reason to use `putIfAbsent`. `put` should be enough for initial map population.

dreab8 commented on hibernate-core/src/main/java/org/hibernate/engine/internal/StatefulPersistenceContext.java:
```diff
@@ -408,7 +408,7 @@ public Object getEntity(EntityKey key) {
 	@Override
 	public boolean containsEntity(EntityKey key) {
-		return entitiesByKey == null ? false : entitiesByKey.containsKey( key );
+		return entitiesByKey != null && entitiesByKey.containsKey( key );
```
just my personal opinion but not sure this helps code readability

NathanQingyangXu replied:
I copied the suggestion by Intellij IDE and I think its suggestion does make sense in this case. If you think the code change is unnecessary, I can revert it back.

dreab8 commented on hibernate-core/src/main/java/org/hibernate/cfg/AbstractPropertyHolder.java:
```diff
@@ -134,7 +134,7 @@ protected ConverterDescriptor makeAttributeConverterDescriptor(AttributeConversi
 	@Override
 	public boolean isInIdClass() {
-		return isInIdClass != null ? isInIdClass : parent != null ? parent.isInIdClass() : false;
+		return isInIdClass != null ? isInIdClass : ( parent != null && parent.isInIdClass() );
```
just my personal opinion but not sure this helps code readability

gavinking replied:
FWIW I think using `&&` instead of a nested `?:` is a lot more readable. However, the parens are unnecessary. `?:` has a lower precedence than `&&`, so I would write:

```jav`
return isInIdClass != null ? isInIdClass : parent != null && parent.isInIdClass();
```

dreab8 replied:
what about using  an If statement instead?
```
if ( isInIdClass != null ) {
   return isInIdClass;
}
else if ( parent != null ) {
   return parent.isInIdClass();
}
else {
   return false;
}```

NathanQingyangXu replied:
> FWIW I think using `&&` instead of a nested `?:` is a lot more readable. However, the parens are unnecessary. `?:` has a lower precedence than `&&`, so I would write:
> 
> ```java
> return isInIdClass != null ? isInIdClass : parent != null && parent.isInIdClass();
> ```

Yeah, technically we don't need parenthesis. I used it to improve the 'perceived complexity' from human code reader. I agree with you generally speaking `&&` is better, but in this case I think Andrea's suggestion makes sense and I am gonna accept his suggestion.

sebersole replied:
IMO as soon as the ternary becomes in any way "non-simple" we ought to use if/else.  So I agree with Andrea's suggestion as well.

sebersole commented:
Overall... I think we ought to consider delaying these types of changes until we get 6.0 more stable.  We still frequently need to merge from master and changes like these are only going to make those processes unnecessarily more difficult.

NathanQingyangXu commented:
> Overall... I think we ought to consider delaying these types of changes until we get 6.0 more stable. We still frequently need to merge from master and changes like these are only going to make those processes unnecessarily more difficult.

This PR target wip/6.0, not master branch. I avoid the issue you mentioned intentionally by working on wip/6.0.

sebersole commented:
Does not matter.  You are (for no logic reasons) creating a merge conflict
that does not need to be there.  That's my point.

Yes, your change is to 6.0.  But what happens when we merge master into 6.0
and Git encounters these diffs?

On Mon, Mar 2, 2020 at 7:17 AM Nathan Xu <notifications@github.com> wrote:

> Overall... I think we ought to consider delaying these types of changes
> until we get 6.0 more stable. We still frequently need to merge from master
> and changes like these are only going to make those processes unnecessarily
> more difficult.
>
> This PR target wip/6.0, not master branch. I avoid the issue you mentioned
> intentionally by working on wip/6.0.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/hibernate/hibernate-orm/pull/3276?email_source=notifications&email_token=AABZIE22X33CY7PLUFVPDIDRFOWU3A5CNFSM4K7GFKF2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOENPIF3I#issuecomment-593396461>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AABZIE4VAYCDXKNIXCSYUOLRFOWU3ANCNFSM4K7GFKFQ>
> .
>

NathanQingyangXu commented:
> Does not matter. You are (for no logic reasons) creating a merge conflict that does not need to be there. That's my point. Yes, your change is to 6.0. But what happens when we merge master into 6.0 and Git encounters these diffs?
> […](#)
> On Mon, Mar 2, 2020 at 7:17 AM Nathan Xu ***@***.***> wrote: Overall... I think we ought to consider delaying these types of changes until we get 6.0 more stable. We still frequently need to merge from master and changes like these are only going to make those processes unnecessarily more difficult. This PR target wip/6.0, not master branch. I avoid the issue you mentioned intentionally by working on wip/6.0. — You are receiving this because you commented. Reply to this email directly, view it on GitHub <#3276?email_source=notifications&email_token=AABZIE22X33CY7PLUFVPDIDRFOWU3A5CNFSM4K7GFKF2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOENPIF3I#issuecomment-593396461>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AABZIE4VAYCDXKNIXCSYUOLRFOWU3ANCNFSM4K7GFKFQ> .

Okay, I am perfectly okay for your decision. No worries. Andrea explained to me your working procedure.  I've thought v6 is rebased on v5, now I know it is more complex than that.

sebersole commented:
@dreab8 Since you so far have been the one doing the merge from master, what are your thoughts here?

dreab8 commented:
@sebersole I have your same concerns, anyway rebasing often as I'm doing should mitigate the problem plus for such changes probably conflict fix should be pretty easy

NathanQingyangXu commented:
> @sebersole I have your same concerns, anyway rebasing often as I'm doing should mitigate the problem plus for such changes probably conflict fix should be pretty easy

I squashed the commits and rebased on wip/6.0 to streamline your processing. Thanks.
